src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,573 @@
+
+/** bitmasks to get the frequency out of a frequency-id */
+const int FREQUENCY_BITMASK = 0xFFF0;
+
+/** bitmasks to get the channel out of a frequency-id */
+const int CHANNEL_BITMASK = 0xF;
+
+class AR62xxDevice final : public AbstractDevice
+{
+
+public:
+  explicit AR62xxDevice(Port &_port);
+
+private:
+  /** Port the radio is connected to. */
+  Port &port;
+
+  /** active station frequency*/
+  RadioFrequency active_frequency;
+
+  /** passive (or standby) station frequency*/
+  RadioFrequency passive_frequency;
+
+  /**
+   * @brief Send - Sends a message to the AR62xx
+   * 
+   * @param msg message
+   * @param msg_size size of the message
+   * @param env operation environment
+   * @return true if message has been sent
+   * @return false if message could not be sent
+   */
+  bool Send(const uint8_t *msg, unsigned msg_size, OperationEnvironment &env);
+
+  /**
+   * @brief converts a frequency to an AR62xx-ID
+   * 
+   * explanation: the AR62xx uses an integer value for each 8.33/25 kHz frequency starting at 118.000 kHz
+   * e.g. frequency-id 0    => 118.000 kHz
+   *      frequency-id 1    => 118.005 kHz
+   *      frequency-id 3040 => 137.000 kHz
+   * 
+   * @param frequency frequency in kHz
+   * @return uint16_t frequency-id
+   */
+  uint16_t FrequencyToAR62xxId(RadioFrequency frequency);
+
+  /**
+   * @brief converts an AR62xx-ID to a real frequency in kHz
+   * 
+   * @param frequency_id frequency-id
+   * @return RadioFrequency frequency in kHz
+   */
+  RadioFrequency AR62xxIdToFrequency(uint16_t frequency_id);
+
+  /**
+   * @brief generates a command to set a station in the AR62xx
+   * 
+   * @param command command-output to set the station
+   * @param station_id the id of the station which should be set
+   *        1 = active station of AR62xx
+   *        0 = passive station of AR62xx
+   * @param frequency frequency in kHz to set
+   * @param station name of the station to set
+   *        (unused)
+   * @return int command-length
+   */
+  int GenerateSetStationCommand(uint8_t *command, int station_id, RadioFrequency frequency, const TCHAR *station);
+
+  /**
+   * @brief parses the received data
+   * this is needed to save the active and passive station
+   * 
+   * @param string recived data
+   * @param len length of data
+   * @return true if data could be parsed
+   * @return false if data could not be parsed
+   */
+  bool ParseReceivedData(const char *string, size_t len);
+
+  /**
+   * @brief save frequencies
+   * 
+   * @param sz_command command which contains the frequencies
+   * @param len length of the command
+   * @param crc CRC
+   */
+  void SaveAR62xxFrequencies(uint8_t *sz_command, int len, uint16_t crc);
+
+  /**
+   * @brief Get the n-th Byte of an unsigned integer
+   * 
+   * @param number unsigned integer
+   * @param n byte-number
+   * @return uint8_t byte
+   */
+  uint8_t GetNthByte(unsigned number, int n);
+
+  /**
+   * @brief Converts to uint8_t-bytes to an unsigned integer
```

Comment:
I don't understand.

Reply:
sorry, there was a typo... anyhow refinined the comment